repose-aggregator/components/filters/cors-filter/src/main/scala/org/openrepose/filters/cors/CorsFilter.scala

```diff
@@ -178,23 +204,71 @@ class CorsFilter @Inject()(configurationService: ConfigurationService)
         Nil
     })
   }
+
+  def getHostUri(request: HttpServletRequestWrapper): URI = {
+    val maybeForwardedHost = request.getSplittableHeaderScala(CommonHttpHeader.X_FORWARDED_HOST).headOption
```

Comment:
Good catch, but it's a moot point.

1. This filter rejecting a request shouldn't be relied upon for anything at all.
2. Leaving off the `Origin` header has the same effect as making the `Origin` header match the `Host` header and as you mentioned making it match the `X-Forwarded-Host` header, although...
3. ...the `X-Forwarded-Host` header is different because it's not on the [forbidden list of headers](https://developer.mozilla.org/en-US/docs/Glossary/Forbidden_header_name) that you can set in a request within a web browser.  Luckily, this is a useless superpower.  If a request tries to set the `X-Forwarded-Host` header which is not a [CORS simple header](https://www.w3.org/TR/cors/#simple-header), the web browser will first send a CORS Preflight Request to ask the third-party server if it's allowed to send the `X-Forwarded-Host` header.  It will do so with the original Origin and Host header which would be rejected by the CORS filter.